posts/2021-05-07-caught-red-handed.md

```diff
@@ -0,0 +1,125 @@
+---
+layout: post
+title: "1.52.0, fingerprints and compiler error forensics"
+author: Felix Klock
+team: the compiler team <https://www.rust-lang.org/governance/teams/compiler>
+---
+
+The Rust teams are always excited to report on new features offered with each release. Sometimes, however, an important change that is not yet "fully baked" gets accidentally included in a release.
```

Comment:
I think it would be really helpful to have a Summary/TL;DR section at the top that says something along the lines of:

- There are bugs in incremental compilation in 1.52.0 stable. 
- Incremental compilation is off by default for release builds, so no production builds should be affected. 
- If you are encountering these bugs during development, you can disable incremental compilation by using the environment variable `CARGO_INCREMENTAL=0` or editing your `$HOME/.cargo/config.toml` or `$HOME/.cargo/config.toml` file to add `build.incremental = 'false'`.
- We're expecting to have a 1.52.1 release on [whatever date]
- We [do|do not] want more bug reports (and if you do want them, what's the most useful information to include)

Reply:
> There are bugs in incremental compilation in 1.52.0 stable.

*and prior versions*

Reply:
> There are bugs in incremental compilation in 1.52.0 stable.

Note that there are bugs in incremental compilation ... in essentially all known versions, in 1.52 they're just very visible. I think we should be very clear that downgrading to 1.51 is likely *not* the path users want to take.

Reply:
Is "There are bugs in incremental compilation in all Rust stable versions that were made into compilation errors in 1.52.0 stable." more accurate? I just edited my post to that